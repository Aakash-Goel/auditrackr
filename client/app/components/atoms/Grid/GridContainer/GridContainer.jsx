import React from 'react';
import { object, node, string } from 'prop-types';

import withStyles from '@material-ui/core/styles/withStyles';
import Grid from '@material-ui/core/Grid';

/**
 * Type checking - Define prop types
 * @private
 */
const propTypes = {
  classes: object.isRequired,
  children: node.isRequired,
  className: string,
};

/**
 * Define default prop values
 * @private
 */
const defaultProps = {
  className: '',
};

const style = {
  grid: {
    marginRight: '-15px',
    marginLeft: '-15px',
    width: 'auto',
  },
};

const GridContainer = ({ ...props }) => {
  const { classes, children, className, ...rest } = props;
  return (
    <Grid container {...rest} className={`${classes.grid} ${className}`}>
      {children}
    </Grid>
  );
};

GridContainer.propTypes = propTypes;
GridContainer.defaultProps = defaultProps;

export default withStyles(style)(GridContainer);
