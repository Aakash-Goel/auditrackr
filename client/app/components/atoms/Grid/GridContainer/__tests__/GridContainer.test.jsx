import React from 'react';
import TestRenderer from 'react-test-renderer';

import GridContainer from '../GridContainer';

describe('<GridContainer/>', () => {
  it('renders correctly with empty props', () => {
    const gridContainerWrapper = TestRenderer.create(
      <GridContainer>My content goes here</GridContainer>
    ).toJSON();
    expect(gridContainerWrapper).toMatchSnapshot();
  });
});
