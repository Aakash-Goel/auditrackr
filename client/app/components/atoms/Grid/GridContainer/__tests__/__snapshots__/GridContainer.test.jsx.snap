// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridContainer/> renders correctly with empty props 1`] = `
<div
  className="MuiGrid-root GridContainer-grid-1 MuiGrid-container"
>
  My content goes here
</div>
`;
